source/common/http/http1/codec_impl.cc

```diff
@@ -861,6 +864,58 @@ void ConnectionImpl::onResetStreamBase(StreamResetReason reason) {
   onResetStream(reason);
 }
 
+void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+  // Dump all bool to provide context
+  os << spaces << "Http1::ConnectionImpl " << this << DUMP_MEMBER(dispatching_)
+     << DUMP_MEMBER(dispatching_slice_already_drained_) << DUMP_MEMBER(reset_stream_called_)
+     << DUMP_MEMBER(handling_upgrade_) << DUMP_MEMBER(deferred_end_stream_headers_)
+     << DUMP_MEMBER(strict_1xx_and_204_headers_) << DUMP_MEMBER(processing_trailers_);
+
+  // Dump the first slice of the dispatching buffer and buffered_body if
+  // applicable.
+  auto dumpBuffer = [](auto* instance) {
+    auto slice = instance->frontSlice();
+    return absl::string_view(static_cast<const char*>(slice.mem_), slice.len_);
+  };
+
+  os << DUMP_NULLABLE_MEMBER(current_dispatching_buffer_, dumpBuffer(current_dispatching_buffer_));
+  os << DUMP_MEMBER(buffered_body_, dumpBuffer(&buffered_body_));
+
+  // Dump header parsing state, and any progress on other headers.
+  os << DUMP_MEMBER(header_parsing_state_);
+  if (header_parsing_state_ == HeaderParsingState::Field) {
+    os << DUMP_MEMBER(current_header_field_, current_header_field_.getStringView());
+  } else if (header_parsing_state_ == HeaderParsingState::Value) {
+    os << DUMP_MEMBER(current_header_field_, current_header_field_.getStringView());
+    os << DUMP_MEMBER(current_header_value_, current_header_value_.getStringView());
+  }
+
+  // Dump Child
```

Comment:
I think that the dump for current_dispatching_buffer_ should be done via DUMP_DETAILS.  Consider putting it after the dump of header maps in subclasses. Reasoning: buffer contents can be long and may contain newlines